File: KHE_AuthService/KHE_AuthService/Entities/OrderDetail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace KHE_AuthService.Entities
{
    public class OrderDetail
    {
        public int Id { get; set; }

        public int OrderId { get; set; }

        public int ProductId { get; set; }

        public int Quantity { get; set; }

        public decimal UnitPrice { get; set; }

        public string? FlavorNotes { get; set; }

        public int? GrindingOptionId { get; set; }

        public virtual Order Order { get; set; } = null!;

        public virtual Product Product { get; set; } = null!;
    }
}

File: LapTrinhTichHop_WebCafe/Backend/src/main/java/com/example/Java_Backend/entities/OrderDetail.java
package com.example.Java_Backend.entities;

import jakarta.persistence.*;
import com.example.Java_Backend.entities.Product;
import com.example.Java_Backend.entities.Order;
import lombok.Data;

@Data
public class OrderDetail {
    private Integer Id;

    private Integer OrderId;

    private Integer ProductId;

    private Integer Quantity;

    private Double UnitPrice;

    private String FlavorNotes;

    private Integer GrindingOptionId;

    private String Weight;

    private Product Product;

    private Order Order;
}
